app/_src/mesh/guides/deploy-kong-mesh-using-terraform-and-konnect.md

```diff
@@ -0,0 +1,606 @@
+
+```bash
+kubectl get meshtrafficpermissions.kuma.io -A
+```
+
+should print:
+
+```
+NAMESPACE          NAME
+kong-mesh-system   allow-all-wd5xx76vc44b498c
+```
+
+## Rotating the token
+
+Instead of using `time_offset` resource you can use `time_rotating` resource to rotate the token every minute:
+
+```hcl
+resource "time_rotating" "example" {
+  rotation_minutes = 1
+}
+```
+
+Then, zone system account token needs updating to use that resource.
+We're adding a 15 minutes offset to the rotation time to ensure the token is rotated before it expires:
+
+```hcl
+resource "konnect_system_account_access_token" "zone_system_account_token" {
+  depends_on = [konnect_system_account.zone_system_account, time_rotating.rotate, time_rotating.rotate]
+  account_id = konnect_system_account.zone_system_account.id
+  expires_at = timeadd(time_rotating.rotate.rotation_rfc3339, "15m")
+  name       = konnect_system_account.zone_system_account.name
+}
+```
+
+{% tip %}
+In production environment you should tune the rotation and offset time to your needs.
+{% endtip %}
+
+That will cause the token to be rotated if, a minute passes `terraform apply`.
+If you execute `terraform apply` immediate after you'll see:
+
+```
+No changes. Your infrastructure matches the configuration.
+```
+
+But, if you run it after a minute you will see zone system account token recreated and Kubernetes secret updated:
+
+```
+  # konnect_system_account_access_token.zone_system_account_token must be replaced
+-/+ resource "konnect_system_account_access_token" "zone_system_account_token" {
+      ~ created_at   = "2025-03-13T11:49:42Z" -> (known after apply)
+      ~ expires_at   = "2025-03-13T12:05:41Z" # forces replacement -> (known after apply) # forces replacement
+      ~ id           = "354b4bee-dfbc-4a10-b1e1-9dcaf4382c72" -> (known after apply)
+      + last_used_at = (known after apply)
+        name         = "mesh_fe75caba-be28-4ff8-850b-4f573f836160_zone1"
+      ~ token        = "spat_Qwb8wBAYJdhTn0m1CaHoP7uXtprbN0xpu3TAG5wbAYAgrHh8g" -> (known after apply)
+      ~ updated_at   = "2025-03-13T11:49:42Z" -> (known after apply)
+        # (1 unchanged attribute hidden)
+    }
+
+  # kubernetes_secret.mesh_cp_token will be updated in-place
+  ~ resource "kubernetes_secret" "mesh_cp_token" {
+      ~ data                           = (sensitive value)
+        id                             = "kong-mesh-system/cp-token"
+        # (3 unchanged attributes hidden)
+
+        # (1 unchanged block hidden)
+    }
+```
+
+{% warning %}
+Currently the control-plane will not pick up the new token automatically. You need to manually restart the control-plane to pick up the new token:
+
+```bash
+kubectl rollout restart -n kong-mesh-system deployment kong-mesh-control-plane
+```
+
+This limitation will be lifted in the next release.
+{% endwarning %}
+
+## Cleaning up
+
+Simply run:
+
+```bash
+terraform destroy
+```
+
+This will clean up all the resources that you created.
```

Comment:
Below this we could add a link to https://github.com/Kong/docs.konghq.com/pull/8585/files when it get merged